src/CMakeLists.txt

```diff
@@ -813,11 +813,8 @@ if (NOT WITH_SYSTEM_ROCKSDB)
 
   # We really want to have the CRC32 calculation in RocksDB accelerated
   # with SSE 4.2. For details refer to rocksdb/util/crc32c.cc.
-  if (HAVE_INTEL_SSE4_2)
-    list(APPEND ROCKSDB_CMAKE_ARGS -DCMAKE_CXX_FLAGS=${SIMD_COMPILE_FLAGS})
-  else()
-    list(APPEND ROCKSDB_CMAKE_ARGS -DWITH_SSE42=OFF)
-  endif()
+  list(APPEND ROCKSDB_CMAKE_ARGS -DWITH_SSE42=OFF)	
```

Comment:
simply disabling the SSE42 instruction is not the ideal solution i think. and `WITH_SSE42` was removed from facebook upstream. probably we can 
give https://github.com/ceph/rocksdb/pull/22 a try with the latest upstream master.

Reply:
Kefu, 

This is part #1 of 2 PRs, the other one is: https://github.com/ceph/rocksdb/pull/23 . Instead of forcing all builds with 'sse4.2', and run that build on a machine that won't support it (which is how we got into this issue), we disable it and check/use 'sse4.2' if/when available dynamically/runtime.